tests/test_models.py

```diff
@@ -34,9 +34,53 @@ def test_infeasible_model():
         )
     )
     m = solph.Model(es)
-    with warnings.catch_warnings(record=True) as w:
-        m.solve(solver="cbc")
-        assert "Optimization ended with status" in str(w[0].message)
+    with warnings.catch_warnings(record=False) as w:
+        m.solve(solver="cbc", allow_nonoptimal=True)
+        assert "The solver did not return an optimal solution" in str(
+            w[0].message
+        )
+
+
+def test_infeasible_model_error():
+    es = solph.EnergySystem(timeincrement=[1])
+    bel = solph.buses.Bus(label="bus")
+    es.add(bel)
+    es.add(
+        solph.components.Sink(
+            inputs={bel: solph.flows.Flow(nominal_value=5, fix=[1])}
+        )
+    )
+    es.add(
+        solph.components.Source(
+            outputs={bel: solph.flows.Flow(nominal_value=4, variable_costs=5)}
+        )
+    )
+    m = solph.Model(es)
+    try:
+        m.solve(solver="cbc", allow_nonoptimal=False)
+    except Exception as e:
+        assert "The solver did not return an optimal solution" in str(e)
```

Comment:
The test case is almost the same as before, so this should do:

```suggestion
    with pytest.raises(RuntimeError, match="The solver did not return an optimal solution"):
        m.solve(solver="cbc", allow_nonoptimal=False)
```